ui/pb/progressbar.go

```diff
@@ -124,18 +124,18 @@ func (pb *ProgressBar) String() string {
 
 	filling := ""
 	caret := ""
+	padding := ""
+	// pb.progress() can make progress become NaN, cause filled overflow.
+	// So only do calculation when filled is a safe value.
 	if filled > 0 {
+		filling = strings.Repeat("=", filled)
 		if filled < space {
 			filling = strings.Repeat("=", filled-1)
 			caret = ">"
-		} else {
-			filling = strings.Repeat("=", filled)
 		}
-	}
-
-	padding := ""
-	if space > filled {
-		padding = pb.color.Sprint(strings.Repeat("-", space-filled))
+		if space > filled {
```

Comment:
this is exactly the previous `if` ... also because you removed the else now the output looks like :
```
running (00m20.1s), 0/2 VUs, 16 complete and 0 interrupted iterations
constant_arr_rate [======================================] 0.75 iters/s, 1 out of 1 VUs active done!
shared_iters1 [] 0/0 shared iters among 0 VUs done!
shared_iters2 [======================================] 1/1 shared iters among 1 VUs done!
```

But the problem is elsewhere: as you've mentioned above in the comment if "progress" is NaN we don't have a "progress" because well there is no work do be done, so what do we do with a "progress" bar then?

At this point, I would consider that the problem is not that we panic (which is a **BIG** problem), but the fact that the code doesn't handle the case in which an executor has no work in some special way. 
My opinion is that in this case, we can do 1 of two things:
1. completely drop that executor from the progressbars and just print a message that there will be no work for it. I would consider this to probably be somewhat more confusing - someone could miss that message with a lot of executors (especially if they are logging stuff for debugging) and then it's missing in the place where all the other executors are ... Seems like a bad UX, also probably more code refactoring 

2. we can just write something like:
```golang
       // pb.progress() can make progress become NaN
       if math.IsNaN(progress) {
               return fmt.Sprintf("%s %s %s", left, "this executor have no iterations to do", right)
       }
      if filled > 0 { 
...
```
Obviously the message should be different ... And maybe we can detect this not through `progress`, but I think we need some discussion about what we should do first and then to implement it (cc @imiric , @na-- ), preferably in the issue and not here  ;)

Also please don't pick to work on code that there are currently open PRs that are being worked on. Now you will (again) need to wait for @imiric  to merge his PR and then rebase and merge yours ... or he will need to do that ... obviously, after both of them are done:)

Reply:
@MStoykov 

> this is exactly the previous if ... also because you removed the else now the output looks like :

No, the previous if does not have `filled > 0` condition. When `filled` overflows, it becomes negative, and the condition `space > filled` is true, causing negative passing to strings.Repeat.

Reply:
> completely drop that executor from the progressbars and just print a message that there will be no work for it. I would consider this to probably be somewhat more confusing - someone could miss that message with a lot of executors (especially if they are logging stuff for debugging) and then it's missing in the place where all the other executors are ... Seems like a bad UX, also probably more code refactoring

I'm thinking about (and actual trying this method first) adding validation before any `Executor.Run` but I see some problem then:

 - The segment itself does not have information about its empty or not, in fact, in every executors, we always:

  + Get num VUs (or maxVus) by calling executor's config related methods and attribute, with segment as param.
  + We can always do a check raw way `if numVUs == 0 || numIters == 0`, but it will not scale and repeat for each executors.

Given the fact that each executor currently won't run anything if segment is empty, I propose to fix the issue here first. At least, this is a bug in `ui/pb` itself, since when it does not cover the overflow case.

I prefer current empty progress bar printed, since when I know what's actually happen.

Reply:
> No, the previous if does not have filled > 0 condition. When filled overflows, it becomes negative, and the condition space > filled is true, causing negative passing to strings.Repeat.
I meant the one above this one in the current code ... 

Your "solution" mostly masks the issue instead of ... as I suggested actually specifying that this is a thing that can happen and then having a specific message .. your empty progressbar looks like a bug to me not like there is no work to be done so we draw ... two square brackets.

Again let's move the discussion on what should happen and then how it should work to the [issue](https://github.com/loadimpact/k6/issues/1295).